Sources/Sentry/SentryClient.m

```diff
@@ -231,6 +232,11 @@ - (SentryId *)sendEvent:(SentryEvent *)event
     return SentryId.empty;
 }
 
+- (SentryId *)captureTransaction:(SentryTransaction *)transaction
+{
+    return [self captureEvent:transaction withScope:[[SentryScope alloc] init]];
```

Comment:
this could call either `captureEnvelope` or the transport directly, remember to append those values to the transaction like `release`, `environment`, `tags`, probably from SentryOptions

Reply:
Thats the point. Capture event append all these values already, lets say the SDK grows and more properties are necessary, we would have to change in one place. Do you think the best option it is to make them part ways?

Reply:
at least on Java/Android we've decided to not do it because transactions are supposed to be cheap, if we do captureEvent on Android it runs event processors, beforeSend etc, reads available memory, free storage, etc, it'd be very expensive and also having a bunch of data in the JSON that are not that useful, so we decided going forward adding the missing attributes later on (based on needs), I believe on iOS would be the same.

Reply:
you could still extract a method with the common attributes from captureEvent and captureTransaction and call that method, right, from both places.

Reply:
Since Java can also be used on the server and have hundreds or thousands of transactions per second, that was critical there. In here I suggested we just extend Transaction (like it was done on @HazAT 's SDKs like JS) since it's a client based transaction and also since it's simpler to implement (splitting things have turned out to be a huge head ache on both Java and .NET even). Since no server side SDK here we have less reason to do it so to move faster I'd vote for inhertiting from Event to add the new fields.

Also, captureEvent is what's used on all the test infrastructure. So the captureEnvelope before was hard to test in this repo, but the addded benefit to get all the device context was an added bonus,

Reply:
> splitting things have turned out to be a huge head ache on both Java and .NET even

I'm not aware of that, why's that?

This will for sure generate a new source of issues here and/or Android, it was decided not to do it on Android to be cheap, either on payload or memory footprint, now we re doing the opposite here for no reason, Mobile has limited resources and will offer if used heavily, think about every transaction doing IO to read available storage, memory, etc... I'm against it till we decide to read only the min. and useful data to be sent over with transactions, your call.
On the Android side, issues will come, where is the device context? why do I have it on iOS and not on Android? and the explanation will be actually the opposite.

Reply:
> > splitting things have turned out to be a huge head ache on both Java and .NET even
> 
> I'm not aware of that, why's that?
> 

Without a clear guidence of what should be set to both transaction and events it happened on both .NET and Java that neither added critical data like environment, release, tags. A lot more code was put into splitting things, to deal with mispatches such as TransactionName vs Transaction (string), and it's a diviation from the current spec that is just unknown territory. This had startTimestamp already on the event since v5 so it was already trying to prepare to have Event match both Transaction and Event.

> This will for sure generate a new source of issues here and/or Android, it was decided not to do it on Android to be cheap, either on payload or memory footprint, now we re doing the opposite here for no reason, Mobile has limited resources and will suffer if used heavily, think about every transaction doing IO to read available storage, memory, etc... I'm against it till we decide to read only the min. and useful data to be sent over with transactions, your call.
> On the Android side, issues will come, where is the device context? why do I have it on iOS and not on Android? and the explanation will be actually the opposite.

>  Android to be cheap

It was decided on Spring to be cheap. Android will have a transaction every few seconds or minutes, compared to hundreds of transactions per second on Spring. So the reason is last important here IMHO